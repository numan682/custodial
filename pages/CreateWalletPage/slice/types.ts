/* --- STATE --- */
export interface CreateWalletState {
  mnemonic: string[]
  checkbox: boolean
}
