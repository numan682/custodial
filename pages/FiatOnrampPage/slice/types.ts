export interface FiatOnrampState {
  thirdPartyAcknowledged: boolean
}
