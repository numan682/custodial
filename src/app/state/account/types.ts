import { Transaction } from 'app/state/transaction/types'
import { ErrorPayload } from 'types/errors'
import { StringifiedBigInt } from 'types/StringifiedBigInt'

export interface BalanceDetails {
  available: StringifiedBigInt | null
  /** This is delayed in getAccount by 20 seconds on oasisscan and 5 seconds on oasismonitor. */
  debonding: StringifiedBigInt | null
  /** This is delayed in getAccount by 20 seconds on oasisscan and 5 seconds on oasismonitor. */
  delegations: StringifiedBigInt | null
  /** This is delayed in getAccount by 20 seconds on oasisscan and 5 seconds on oasismonitor. */
  total: StringifiedBigInt | null
}

export interface Allowance {
  address: string
  amount: StringifiedBigInt
}

export interface Account extends BalanceDetails {
  address: string
  allowances?: Allowance[]
}

/* --- STATE --- */
export interface AccountState extends Account {
  loading: boolean
  accountError?: ErrorPayload
  transactions: Transaction[]
  transactionsError?: ErrorPayload
}
