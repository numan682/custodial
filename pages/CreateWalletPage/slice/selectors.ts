import { createSelector } from '@reduxjs/toolkit'

import { RootState } from 'types'
import { initialState } from '.'

const selectSlice = (state: RootState) => state.createWallet || initialState

export const selectCreateWallet = createSelector([selectSlice], state => state)
export const selectMnemonic = createSelector([selectCreateWallet], wallet => wallet.mnemonic)
export const selectCheckbox = createSelector([selectCreateWallet], wallet => wallet.checkbox)
