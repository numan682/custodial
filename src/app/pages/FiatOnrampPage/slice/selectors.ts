import { createSelector } from '@reduxjs/toolkit'

import { RootState } from 'types'
import { initialState } from '.'

const selectSlice = (state: RootState) => state.fiatOnramp || initialState

export const selectFiatOnramp = createSelector([selectSlice], state => state)
export const selectThirdPartyAcknowledged = createSelector(
  [selectFiatOnramp],
  wallet => wallet.thirdPartyAcknowledged,
)
