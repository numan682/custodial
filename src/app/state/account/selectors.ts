import { createSelector } from '@reduxjs/toolkit'

import { RootState } from 'types'
import { initialState } from '.'

const selectSlice = (state: RootState) => state.account || initialState

export const selectAccount = createSelector([selectSlice], state => state)
export const selectTransactions = createSelector([selectAccount], account => account.transactions)
export const selectTransactionsError = createSelector([selectAccount], account => account.transactionsError)
export const selectAccountAddress = createSelector([selectAccount], account => account.address)
export const selectAccountAvailableBalance = createSelector([selectAccount], account => account.available)
export const selectAccountIsLoading = createSelector([selectAccount], account => account.loading)
export const selectAccountAllowances = createSelector([selectAccount], account => account.allowances)
