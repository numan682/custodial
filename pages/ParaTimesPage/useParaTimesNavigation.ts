import { useCallback } from 'react'
import { useDispatch, useSelector } from 'react-redux'
import { useTranslation } from 'react-i18next'
import { paraTimesActions } from 'app/state/paratimes'
import { backend } from 'vendors/backend'
import { BackendAPIs } from 'config'
import { WalletType } from 'app/state/wallet/types'
import { selectType } from 'app/state/wallet/selectors'

export type ParaTimesNavigationHook = {
  canAccessParaTimesRoute: boolean
  getParaTimesRoutePath: (address: string) => string
  navigateToAmount: () => void
  navigateToDeposit: () => void
  navigateToParaTimes: () => void
  navigateToRecipient: () => void
  navigateToConfirmation: () => void
  navigateToSummary: () => void
  navigateToWithdraw: () => void
  paraTimesRouteLabel: string
}

export const useParaTimesNavigation = (): ParaTimesNavigationHook => {
  const { t } = useTranslation()
  const dispatch = useDispatch()
  const walletType = useSelector(selectType)
  const canAccessParaTimesRoute = backend() === BackendAPIs.OasisScan && walletType !== WalletType.Ledger
  const getParaTimesRoutePath = (address: string) => `/account/${address}/paratimes`
  const navigateToDeposit = useCallback(() => dispatch(paraTimesActions.navigateToDeposit()), [dispatch])
  const navigateToWithdraw = useCallback(() => dispatch(paraTimesActions.navigateToWithdraw()), [dispatch])
  const navigateToParaTimes = useCallback(() => dispatch(paraTimesActions.navigateToParaTimes()), [dispatch])
  const navigateToRecipient = useCallback(() => dispatch(paraTimesActions.navigateToRecipient()), [dispatch])
  const navigateToAmount = useCallback(() => dispatch(paraTimesActions.navigateToAmount()), [dispatch])
  const navigateToConfirmation = useCallback(
    () => dispatch(paraTimesActions.navigateToConfirmation()),
    [dispatch],
  )
  const navigateToSummary = useCallback(() => dispatch(paraTimesActions.navigateToSummary()), [dispatch])
  const paraTimesRouteLabel = t('menu.paraTimes', 'ParaTimes')

  return {
    canAccessParaTimesRoute,
    getParaTimesRoutePath,
    navigateToAmount,
    navigateToConfirmation,
    navigateToDeposit,
    navigateToParaTimes,
    navigateToRecipient,
    navigateToSummary,
    navigateToWithdraw,
    paraTimesRouteLabel,
  }
}
