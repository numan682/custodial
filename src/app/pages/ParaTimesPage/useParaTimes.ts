import { useCallback } from 'react'
import { useDispatch, useSelector } from 'react-redux'
import { useTranslation } from 'react-i18next'
import { TFunction } from 'i18next'
import { paraTimesActions } from 'app/state/paratimes'
import { TransactionForm, TransactionTypes } from 'app/state/paratimes/types'
import { selectSelectedNetwork, selectTicker } from 'app/state/network/selectors'
import { selectAccountAvailableBalance, selectAccountIsLoading } from 'app/state/account/selectors'
import { selectAddress } from 'app/state/wallet/selectors'
import { selectParaTimes } from 'app/state/paratimes/selectors'
import { StringifiedBigInt } from 'types/StringifiedBigInt'
import { ErrorPayload, ExhaustedTypeError } from 'types/errors'
import { paraTimesConfig, RuntimeTypes, ParaTime, type ParaTimeConfig } from '../../../config'

type AvailableParaTimesForNetwork = {
  isEvm: boolean
  value: ParaTime
}

const evmcParaTimes = Object.keys(paraTimesConfig).filter(
  key => paraTimesConfig[key as ParaTime].type === RuntimeTypes.Evm,
)

export type ParaTimesHook = {
  accountAddress: string | undefined
  accountIsLoading: boolean
  availableParaTimesForSelectedNetwork: AvailableParaTimesForNetwork[]
  balance: StringifiedBigInt | null
  balanceInBaseUnit: boolean
  clearTransactionForm: () => void
  isDepositing: boolean
  isEvmcParaTime: boolean
  isLoading: boolean
  isWalletEmpty: boolean
  paraTimeConfig: ParaTimeConfig
  paraTimeName: string
  setTransactionForm: (formValues: TransactionForm) => void
  submitTransaction: () => void
  ticker: string
  transactionError: ErrorPayload | undefined
  transactionForm: TransactionForm
  usesOasisAddress: boolean
}

const getParaTimeName = (t: TFunction, paraTime: ParaTime) => {
  switch (paraTime) {
    case ParaTime.Cipher:
      return t('paraTimes.common.cipher', 'Cipher')
    case ParaTime.Emerald:
      return t('paraTimes.common.emerald', 'Emerald')
    case ParaTime.Sapphire:
      return t('paraTimes.common.sapphire', 'Sapphire')
    default:
      throw new ExhaustedTypeError(
        t('paraTimes.validation.unsupportedParaTime', 'Unsupported ParaTime'),
        paraTime,
      )
  }
}

export const useParaTimes = (): ParaTimesHook => {
  const { t } = useTranslation()
  const dispatch = useDispatch()
  const clearTransactionForm = useCallback(() => {
    dispatch(paraTimesActions.clearTransactionForm())
  }, [dispatch])
  const submitTransaction = useCallback(() => {
    dispatch(paraTimesActions.submitTransaction())
  }, [dispatch])
  const setTransactionForm = (formValues: TransactionForm) =>
    dispatch(paraTimesActions.setTransactionForm(formValues))
  const { balance, isLoading, transactionError, transactionForm } = useSelector(selectParaTimes)
  const accountBalance = useSelector(selectAccountAvailableBalance)
  const accountIsLoading = useSelector(selectAccountIsLoading)
  const accountAddress = useSelector(selectAddress)
  const selectedNetwork = useSelector(selectSelectedNetwork)
  const ticker = useSelector(selectTicker)
  const isDepositing = transactionForm.type !== TransactionTypes.Withdraw
  const isEvmcParaTime = evmcParaTimes.includes(transactionForm.paraTime!)
  const needsEthAddress = isDepositing && isEvmcParaTime
  const balanceInBaseUnit = isDepositing || (!isDepositing && !isEvmcParaTime)
  const paraTimeName = transactionForm.paraTime ? getParaTimeName(t, transactionForm.paraTime) : ''
  const availableParaTimesForSelectedNetwork: AvailableParaTimesForNetwork[] = (
    Object.keys(paraTimesConfig) as ParaTime[]
  )
    .filter(paraTimeKey => paraTimesConfig[paraTimeKey][selectedNetwork].runtimeId)
    .sort((a, b) => paraTimesConfig[a].displayOrder - paraTimesConfig[b].displayOrder)
    .map(paraTimeKey => ({
      isEvm: paraTimesConfig[paraTimeKey].type === RuntimeTypes.Evm,
      value: paraTimeKey,
    }))
  const walletBalance = !isDepositing ? balance : accountBalance
  const paraTimeConfig = paraTimesConfig[transactionForm.paraTime!]

  return {
    accountAddress,
    accountIsLoading,
    availableParaTimesForSelectedNetwork,
    balance: walletBalance,
    balanceInBaseUnit,
    clearTransactionForm,
    isDepositing,
    isEvmcParaTime,
    isLoading,
    isWalletEmpty: walletBalance === '0',
    paraTimeConfig,
    paraTimeName,
    setTransactionForm,
    submitTransaction,
    ticker,
    transactionError,
    transactionForm,
    usesOasisAddress: !needsEthAddress,
  }
}
