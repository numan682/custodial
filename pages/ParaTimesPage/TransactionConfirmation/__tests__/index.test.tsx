import * as React from 'react'
import { render, screen } from '@testing-library/react'
import { useSelector } from 'react-redux'
import { when } from 'jest-when'
import userEvent from '@testing-library/user-event'
import { TransactionForm, TransactionTypes } from 'app/state/paratimes/types'
import { selectValidators } from 'app/state/staking/selectors'
import { selectWalletsAddresses } from 'app/state/wallet/selectors'
import { useParaTimes, ParaTimesHook } from '../../useParaTimes'
import { useParaTimesNavigation, ParaTimesNavigationHook } from '../../useParaTimesNavigation'
import { TransactionConfirmation } from '..'

jest.unmock('react-i18next')
jest.mock('../../useParaTimes')
jest.mock('../../useParaTimesNavigation')
jest.mock('react-redux', () => ({
  useSelector: jest.fn(),
}))

describe('<TransactionConfirmation />', () => {
  const mockUseParaTimesResult = {
    isDepositing: true,
    isEvmcParaTime: false,
    paraTimeName: 'Cipher',
    ticker: 'ROSE',
    transactionForm: {
      amount: '10',
      recipient: 'dummyAddress',
    },
    usesOasisAddress: true,
  } as ParaTimesHook
  const mockUseParaTimesNavigationResult = {} as ParaTimesNavigationHook

  beforeEach(() => {
    jest.mocked(useParaTimes).mockReturnValue(mockUseParaTimesResult)
    jest.mocked(useParaTimesNavigation).mockReturnValue(mockUseParaTimesNavigationResult)
    when(useSelector as any)
      .calledWith(selectValidators)
      .mockReturnValue([])
      .calledWith(selectWalletsAddresses)
      .mockReturnValue(['dummyAddress'])
  })

  it('should render component', () => {
    const { container } = render(<TransactionConfirmation />)

    expect(container).toMatchSnapshot()
  })

  it('should render EVMc withdraw variant component', () => {
    jest.mocked(useParaTimes).mockReturnValue({
      isDepositing: false,
      isEvmcParaTime: true,
      paraTimeName: 'Emerald',
      ticker: 'ROSE',
      transactionForm: {
        amount: '10',
        recipient: 'dummyAddress',
        type: TransactionTypes.Withdraw,
      },
    } as ParaTimesHook)
    render(<TransactionConfirmation />)

    expect(screen.getByTestId('paraTime-content-description')).toMatchSnapshot()
  })

  it('should require a confirmation on form submit', async () => {
    const navigateToSummary = jest.fn()
    jest
      .mocked(useParaTimesNavigation)
      .mockReturnValue({ ...mockUseParaTimesNavigationResult, navigateToSummary })
    render(<TransactionConfirmation />)

    await userEvent.click(screen.getByRole('button', { name: 'Deposit' }))

    expect(screen.getByText('Field is required')).toBeInTheDocument()
    expect(navigateToSummary).not.toHaveBeenCalled()
  })

  it('should render additional confirmation checkbox when transferring tokens to validator', async () => {
    jest.mocked(useParaTimes).mockReturnValue({
      ...mockUseParaTimesResult,
      transactionForm: {
        ...mockUseParaTimesResult.transactionForm,
        recipient: 'validatorAddress',
      },
    } as ParaTimesHook)
    when(useSelector as any)
      .calledWith(selectValidators)
      .mockReturnValue([{ address: 'validatorAddress' }])
    render(<TransactionConfirmation />)

    expect(screen.getByText('I confirm I want to transfer ROSE to a validator address')).toBeInTheDocument()
  })

  it('should render additional confirmation checkbox when transferring tokens to foreign account', async () => {
    when(useSelector as any)
      .calledWith(selectWalletsAddresses)
      .mockReturnValue(['addressInAccount'])
    render(<TransactionConfirmation />)

    expect(
      screen.getByText('I confirm I want to directly withdraw to an external account'),
    ).toBeInTheDocument()
  })

  it('should submit transaction', async () => {
    const submitTransaction = jest.fn()
    jest.mocked(useParaTimes).mockReturnValue({
      ...mockUseParaTimesResult,
      transactionForm: {
        amount: '10',
        confirmTransfer: true,
        recipient: 'dummyAddress',
        type: TransactionTypes.Withdraw,
      } as TransactionForm,
      submitTransaction,
    })
    render(<TransactionConfirmation />)

    await userEvent.click(screen.getByRole('button', { name: 'Deposit' }))

    expect(submitTransaction).toHaveBeenCalled()
  })

  it('should navigate back to amount selection step', async () => {
    const navigateToAmount = jest.fn()
    jest.mocked(useParaTimesNavigation).mockReturnValue({
      ...mockUseParaTimesNavigationResult,
      navigateToAmount,
    })
    render(<TransactionConfirmation />)

    await userEvent.click(screen.getByRole('button', { name: 'Back' }))

    expect(navigateToAmount).toHaveBeenCalled()
  })
})
