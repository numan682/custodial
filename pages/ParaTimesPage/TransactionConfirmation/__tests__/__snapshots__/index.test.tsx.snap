// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TransactionConfirmation /> should render EVMc withdraw variant component 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  text-align: center;
}

@media only screen and (max-width:768px) {

}

@media only screen and (max-width:768px) {

}

@media only screen and (max-width:768px) {

}

@media only screen and (max-width:768px) {

}

@media only screen and (max-width:768px) {

}

@media only screen and (max-width:768px) {

}

@media only screen and (max-width:768px) {

}

@media only screen and (max-width:768px) {

}

@media only screen and (max-width:768px) {

}

@media only screen and (max-width:768px) {

}

@media only screen and (max-width:768px) {

}

@media only screen and (max-width:768px) {

}

<span
  class="c0"
  data-testid="paraTime-content-description"
>
  <span>
    You are about to withdraw 
  </span>
  <strong>
    <span>
      10 ROSE
    </span>
  </strong>
  <span>
     from 
  </span>
  <strong>
    <span>
      Emerald
    </span>
  </strong>
  <span>
     (EVMc) to 
  </span>
  <strong>
    <span>
      dummyAddress
    </span>
  </strong>
</span>
`;

exports[`<TransactionConfirmation /> should render component 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  margin-bottom: 48px;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  margin-bottom: 96px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  margin-bottom: 24px;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  margin-bottom: 12px;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  background-color: alert-box-warning-background;
  color: #444444;
  border-left: solid 3px alert-box-warning;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding-left: 12px;
  padding-right: 12px;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

.c10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  margin-bottom: 12px;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  border-bottom: solid 1px rgba(0,0,0,0.33);
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 12px;
}

.c13 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  margin-right: 12px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c16 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: solid 2px rgba(0,0,0,0.15);
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 24px;
  width: 24px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border-radius: 4px;
}

.c17 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  margin-bottom: 48px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c12 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  cursor: pointer;
}

.c12:hover input:not([disabled]) + div,.c12:hover input:not([disabled]) + span {
  border-color: #000000;
}

.c15 {
  opacity: 0;
  -moz-appearance: none;
  width: 0;
  height: 0;
  margin: 0;
  cursor: pointer;
}

.c15:checked + span > span {
  left: calc( 48px - 24px );
  background: #7D4CDB;
}

.c14 {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.c3 {
  font-size: 18px;
  line-height: 24px;
  text-align: center;
}

.c9 {
  font-size: 12px;
  line-height: normal;
  font-weight: bold;
}

.c18 {
  display: inline-block;
  box-sizing: border-box;
  cursor: pointer;
  font: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  margin: 0;
  background: transparent;
  overflow: visible;
  text-transform: none;
  border: 2px solid #7D4CDB;
  border-radius: 18px;
  color: #444444;
  padding: 4px 22px;
  font-size: 18px;
  line-height: 24px;
  background-color: #7D4CDB;
  color: #f8f8f8;
  border-radius: 18px;
  -webkit-transition-property: color,background-color,border-color,box-shadow;
  transition-property: color,background-color,border-color,box-shadow;
  -webkit-transition-duration: 0.1s;
  transition-duration: 0.1s;
  -webkit-transition-timing-function: ease-in-out;
  transition-timing-function: ease-in-out;
  width: 100%;
}

.c18:hover {
  box-shadow: 0px 0px 0px 2px #7D4CDB;
}

.c18:focus {
  outline: none;
  box-shadow: 0 0 2px 2px #6FFFB0;
}

.c18:focus > circle,
.c18:focus > ellipse,
.c18:focus > line,
.c18:focus > path,
.c18:focus > polygon,
.c18:focus > polyline,
.c18:focus > rect {
  outline: none;
  box-shadow: 0 0 2px 2px #6FFFB0;
}

.c18:focus::-moz-focus-inner {
  border: 0;
}

.c18:focus:not(:focus-visible) {
  outline: none;
  box-shadow: none;
}

.c18:focus:not(:focus-visible) > circle,.c18:focus:not(:focus-visible) > ellipse,
.c18:focus:not(:focus-visible) > line,.c18:focus:not(:focus-visible) > path,
.c18:focus:not(:focus-visible) > polygon,.c18:focus:not(:focus-visible) > polyline,
.c18:focus:not(:focus-visible) > rect {
  outline: none;
  box-shadow: none;
}

.c18:focus:not(:focus-visible)::-moz-focus-inner {
  border: 0;
}

.c1 {
  margin-bottom: 48px;
  font-size: 26px;
  line-height: 32px;
  max-width: 624px;
  font-weight: 600;
  overflow-wrap: break-word;
}

@media only screen and (max-width:768px) {
  .c2 {
    margin-bottom: 24px;
  }
}

@media only screen and (max-width:768px) {
  .c5 {
    margin-bottom: 12px;
  }
}

@media only screen and (max-width:768px) {
  .c7 {
    border-left: solid 3px alert-box-warning;
  }
}

@media only screen and (max-width:768px) {
  .c7 {
    padding-left: 6px;
    padding-right: 6px;
  }
}

@media only screen and (max-width:768px) {
  .c10 {
    margin-bottom: 6px;
  }
}

@media only screen and (max-width:768px) {
  .c11 {
    border-bottom: solid 1px rgba(0,0,0,0.33);
  }
}

@media only screen and (max-width:768px) {
  .c11 {
    padding: 6px;
  }
}

@media only screen and (max-width:768px) {
  .c13 {
    margin-right: 6px;
  }
}

@media only screen and (max-width:768px) {
  .c16 {
    border: solid 2px rgba(0,0,0,0.15);
  }
}

@media only screen and (max-width:768px) {
  .c17 {
    margin-bottom: 24px;
  }
}

@media only screen and (max-width:768px) {
  .c1 {
    margin-bottom: 24px;
  }
}

@media only screen and (max-width:768px) {
  .c1 {
    font-size: 18px;
    line-height: 24px;
    max-width: 432px;
  }
}

<div>
  <div
    class="c0"
  >
    <h3
      class="c1"
    >
      Deposit to ParaTime
    </h3>
    <div
      class="c2"
      style="max-width: 550px;"
    >
      <span
        class="c3"
        data-testid="paraTime-content-description"
      >
        <span>
          You are about to deposit 
        </span>
        <strong>
          <span>
            10 ROSE
          </span>
        </strong>
        <span>
           to 
        </span>
        <strong>
          <span>
            dummyAddress
          </span>
        </strong>
        <span>
           on 
        </span>
        <strong>
          <span>
            Cipher
          </span>
        </strong>
        <span>
           
        </span>
      </span>
    </div>
    <div
      class="c4"
    >
      <form>
        <div
          class="c5"
          style="max-width: 450px;"
        >
          <div
            class="c6"
          >
            <div
              class="c7"
            >
              <div
                class="c8"
              >
                <span
                  class="c9"
                  style="margin-top: 10px; margin-bottom: 10px;"
                >
                  <span
                    class="c3"
                  >
                    Please confirm the deposit amount and the recipient address are correct and then click "Deposit".
                  </span>
                </span>
              </div>
            </div>
          </div>
          <div
            class="c10 "
          >
            <div
              class="c11 "
            >
              <label
                class="c12"
                label="I confirm the amount and the address are correct"
              >
                <div
                  class="c13 c14"
                >
                  <input
                    class="c15"
                    name="confirmTransfer"
                    type="checkbox"
                  />
                  <div
                    class="c16 "
                  />
                </div>
                <span>
                  I confirm the amount and the address are correct
                </span>
              </label>
            </div>
          </div>
        </div>
        <div
          class="c17"
        >
          <div
            class="c5"
          >
            <button
              class="c18"
              type="submit"
            >
              Deposit
            </button>
          </div>
        </div>
      </form>
    </div>
  </div>
</div>
`;
